ci/doc_only_change.sh

```diff
@@ -9,10 +9,10 @@ mb_modifier=
 if [ "$CHANGE_ID" = "null" ]; then
     mb_modifier=^
```

Comment:
Alternative short change to fixing the Jenkinsfile would be for line 10 to return 0.   As we really do not want to skip builds on a non-PR branch.  It results in a successful build with no artifacts, which breaks other tools that try to get artifacts from the last successful build.

Reply:
> The problem is that because the results are not cached (bug in Jenkinsfile) the tip of master is moving.

That is not actually the problem.  Not caching the results actually resulted in shining a light on the problem.

The problem is using a constantly moving _pointer_ such as a branch name like `master` which provides incoherent and possibly incorrect results.  This makes the use of the branch name racy such that even if you do cache the result of the first query, that query itself could be wrong due to the `master` _pointer_ having moved (due to another landing in quick succession) since the job that started.

But as @JohnMalmberg says, the fix proposed here is not the right fix.